cmd/main.go

```diff
@@ -1,124 +1,86 @@
-	price, err := postgres.FetchPrice(claims.Item)
-	if err != nil {
-		log.Printf("fetchPrice %v", err)
-		return
+func newPriceHandler() (http.HandlerFunc, error) {
+	remoteServer := ""
+	if s := os.Getenv("REMOTE_SERVER"); s == "" {
+		return nil, fmt.Errorf("env var REMOTE_SERVER not found")
+	} else {
+		remoteServer = s
 	}
+	dsClient := downstream.New(remoteServer)
+	return func(w http.ResponseWriter, r *http.Request) {
+		// We should always read the body and close it, even if we don't need it. If we don't
+		// we'll leak goroutines and http.Requests....
+		_, err := io.ReadAll(r.Body)
+		if err != nil {
+			log.Print(err)
+		}
+		defer r.Body.Close()
 
-	if price >= 0 {
+		jwtToken := r.URL.Query().Get("jwt")
 
-		discount, err := fetchDiscount(claims.Quantity)
+		claims, err := dsClient.ParseJWT(jwtToken)
 		if err != nil {
-			log.Printf("Fetch Discount %v", err)
+			log.Printf("Error parsing JWT: %v", err)
+			w.WriteHeader(http.StatusBadRequest)
 			return
 		}
 
-		var totalPrice float32
-		var discountedPrice float32 = float32(price) - float32(price)*(discount.Discount)/float32(100)
-		totalPrice = float32(discountedPrice * float32(claims.Quantity))
-
-		if claims.Vatincl {
-			totalPrice = totalPrice * 1.2
+		// Fetch the price from Postgres
+		price, err := postgres.FetchPrice(claims.Item)
+		if err != nil {
+			log.Printf("fetchPrice %v", err)
+			return
 		}
 
-		// Set the content type to application/json
-		w.Header().Set("Content-Type", "application/json")
-		// Write the JSON response
-		fmt.Fprintf(w, "{\"quality\":%d,\"tot_price\":%.2f }", claims.Quantity, totalPrice)
-		fmt.Println("Request done")
-	} else {
-		log.Println("Price < 0!")
-		w.WriteHeader(http.StatusInternalServerError)
-	}
-}
-
-func fetchDiscount(quantity int) (DiscountResponse, error) {
-	// The URL of the HTTP server you want to call
-	baseUrl := os.Getenv("REMOTE_SERVER")
-	if len(baseUrl) == 0 {
-		baseUrl = "http://localhost:7070"
-	}
-	baseUrl = baseUrl + "/discount"
-
-	// Create a URL object
-	u, err := url.Parse(baseUrl)
-	if err != nil {
-		log.Fatalf("Error parsing URL: %v\n", err)
-	}
-
-	query := u.Query()
-	query.Set("quantity", strconv.Itoa(quantity))
-	u.RawQuery = query.Encode()
-
-	// Make the GET request
-	resp, err := http.Get(u.String())
-	if err != nil {
-		log.Printf("Error making GET request: %v", err)
-		return DiscountResponse{}, err
-	}
-	defer resp.Body.Close()
-
-	// Check if the response status is 200 OK
-	if resp.StatusCode != http.StatusOK {
-		log.Fatalf("Error: status code is %d\n", resp.StatusCode)
-	}
-
-	// Decode the JSON response directly into the struct
-	var discountResp DiscountResponse
-	decoder := json.NewDecoder(resp.Body)
-	err = decoder.Decode(&discountResp)
-	if err != nil {
-		log.Fatalf("Error decoding JSON: %v\n", err)
-	}
-
-	// Print the unmarshalled data
-	// log.Printf("Item: %s Discount: %.2f Quantity: %d\n", discountResp.Item, discountResp.Discount, discountResp.Quantity)
-	return discountResp, err
```

Comment:
This diff is gnarlier than it is in reality.... Not sure why git is so confused...